src/main/java/org/stellar/sdk/InvokeHostFunctionOperation.java

```diff
@@ -0,0 +1,153 @@
+package org.stellar.sdk;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Objects;
+import java.util.Arrays;
+import org.stellar.sdk.xdr.*;
+
+/**
+ * Represents <a
+ * href="https://developers.stellar.org/docs/fundamentals-and-concepts/list-of-operations#invoke-host-function"
+ * target="_blank">InvokeHostFunction</a> operation.
+ *
+ * @see <a href="https://developers.stellar.org/docs/fundamentals-and-concepts/list-of-operations"
+ *     target="_blank">List of Operations</a>
+ */
+public class InvokeHostFunctionOperation extends Operation {
+
+  private final HostFunction hostFunction;
+  private SorobanAuthorizationEntry[] auth;
+
+  /**
+   * Constructs a new InvokeHostFunctionOperation object using the specified parameters.
+   *
+   * @param hostFunction host function to execute (with its wrapped parameters)
+   * @param auth the authorizations required to execute the host function
+   */
+  private InvokeHostFunctionOperation(HostFunction hostFunction, SorobanAuthorizationEntry[] auth) {
+    this.hostFunction = hostFunction;
+    this.auth = auth;
+  }
+
+  /**
+   * Returns host function to execute (with its wrapped parameters).
+   *
+   * @return The host function to execute.
+   */
+  public HostFunction getHostFunction() {
+    return hostFunction;
+  }
+
+  /**
+   * Returns the authorizations required to execute the host function.
+   *
+   * @return The authorizations.
+   */
+  public SorobanAuthorizationEntry[] getAuth() {
+    return auth;
+  }
+
+  /**
+   * Sets the authorizations required to execute the host function.
+   *
+   * @param auth The authorizations.
+   */
+  public void setAuth(SorobanAuthorizationEntry[] auth) {
+    this.auth = auth;
+  }
+
+  @Override
+  org.stellar.sdk.xdr.Operation.OperationBody toOperationBody(AccountConverter accountConverter) {
+    InvokeHostFunctionOp op = new InvokeHostFunctionOp();
+    op.setHostFunction(this.hostFunction);
+    op.setAuth(this.auth);
+
+    org.stellar.sdk.xdr.Operation.OperationBody body =
+        new org.stellar.sdk.xdr.Operation.OperationBody();
+    body.setDiscriminant(OperationType.INVOKE_HOST_FUNCTION);
+    body.setInvokeHostFunctionOp(op);
+
+    return body;
+  }
+
+  /**
+   * Builds InvokeHostFunction operation.
+   *
+   * @see InvokeHostFunctionOperation
+   */
+  public static class Builder {
```

Comment:
rather than write InvokeHostFunctionOperations as a manual pojo with the setters/getters, etc and this builder, could use lombok `@Value` and `@Builder` on `InvokeHostFunctionOperation` instead to auto-generate bytecode that does the same, checkout TransactionsPreconditions for example usage:

https://github.com/stellar/java-stellar-sdk/blob/soroban/src/main/java/org/stellar/sdk/TransactionPreconditions.java#L20

https://github.com/stellar/java-stellar-sdk/blob/soroban/src/test/java/org/stellar/sdk/TransactionPreconditionsTest.java

this creates an immutable pojo with builder. So, when a client needs a new instance use `InvokeHostFunctionOperation.builder()` or if need to update existing instance, use the `toBuilder()` method on the instance.

Reply:
Fixed in 5838abd4f57793f3e9295ff55817d6d433d1d736